src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -536,4 +539,28 @@
      */
     List<BannerPatternShape> getBannerPatternShapes();
 
+    /**
+     * Gets an {@link Attribute} by name.
+     * 
+     * @param name The name to find
+     * @return An Attribute if one could be found, otherwise
+     *         {@link Optional#absent()}.
+     */
+    Optional<Attribute> getAttribute(String name);
+
+    /**
+     * Gets an {@link Operation} by id.
+     * 
+     * @param id The id to find
+     * @return An Operation if one could be found, otherwise
+     *         {@link Optional#absent()}.
+     */
+    Optional<Operation> getOperation(int id);
```

Comment:
@AlphaModder Maybe a String name similar to an enum name. This would be abstract enough to be useful for storing in a config as well. But wait for an official decision here.